src/ARKit/ARFaceGeometry.cs

```diff
@@ -0,0 +1,34 @@
+//
+// ARFaceGeometry.cs: Nicer code for ARFaceGeometry
+//
+// Authors:
+//	Vincent Dondain  <vidondai@microsoft.com>
+//
+// Copyright 2017 Microsoft Inc. All rights reserved.
+//
+
+#if XAMCORE_2_0
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace XamCore.ARKit {
+	public partial class ARFaceGeometry {
+
+		// Calling this one 'TriangleIndexes' so it doesn't clash with 'TriangleIndices'.
+		// We could use a method here but a property is more consistent with other ARKit APIs like ARPointCloud.
+		public unsafe short [] TriangleIndexes {
+			get {
+				// There are always 3x more 'TriangleIndices' than 'TriangleCount' since 'TriangleIndices' represents Triangles (set of three indices).
+				var count = (int)TriangleCount * 3;
+				var rv = new short [count];
+				var ptr = (short *) GetTriangleIndexes ();
+				for (int i = 0; i < count; i++)
+					rv [i] = *ptr++;
```

Comment:
not sure how many items there is usually but it might be better to use `Marshal.Copy` and drop the unsafe

Reply:
I don't have an exact answer to this question, would need a test case. However this from the documentation: "Each 16-bit integer value in this buffer represents an index into the vertices and textureCoordinates buffers. Each set of three indices identifies the vertices comprising a single triangle in the mesh." makes me think it can be quite a lot (all the meshes representing a face).

Also I'd like to point out that we use the same code for ARPointCloud: https://github.com/xamarin/xamarin-macios/blob/master/src/ARKit/ARPointCloud.cs#L19-L39 (initially suggested by Rolf).